### PR TITLE
DOC, MAINT: more refguide fixes

### PR DESCRIPTION
**Pull request by tylerjereddy**
* see: https://github.com/scipy/scipy/pull/19525#issuecomment-1813073890

* this allows the refguide check to pass for me locally; I use `openblas-0.3.23` from `spack` locally, maybe that's part of the difference

[skip ci] [skip circle]

### CODE REVIEW

File: scipy/linalg/_decomp_schur.py [hunk 1]
```diff
@@ -100,9 +100,9 @@ def schur(a, output='real', lwork=None, overwrite_a=False, sort=None,
 
     >>> T2, Z2 = schur(A, output='complex')
     >>> T2
-    array([[ 2.65896708, -1.22839825+1.32378589j,  0.42590089+1.51937378j],
-           [ 0.        , -0.32948354+0.80225456j, -0.59877807+0.56192146j],
-           [ 0.        ,  0.                    , -0.32948354-0.80225456j]])
+    array([[ 2.65896708, -1.22839825+1.32378589j,  0.42590089+1.51937378j], # may vary
+           [ 0.        , -0.32948354+0.80225456j, -0.59877807+0.56192146j], # may vary
+           [ 0.        ,  0.                    , -0.32948354-0.80225456j]]) # may vary
```
Comment (ev-br) [inline scipy/linalg/_decomp_schur.py:105]: Strictly speaking, a single `# may vary` anywhere in the output is enough, cf https://github.com/scipy/scipy/blob/main/tools/refguide_check.py#L578
(here `want` is lines 103-105, `array([[...]])`)

So to me at least this reads like you're really stressing that each row is essentially random, not just the whole matrix (?) --- is this your intention?

File: scipy/stats/_mstats_extras.py [hunk 2]
```diff
@@ -72,7 +72,7 @@ def hdquantiles(data, prob=list([.25,.5,.75]), axis=None, var=False,):
     >>> # Display the quantile estimates
     >>> for i, quantile in enumerate(probabilities):
     ...     print(f"{int(quantile * 100)}th percentile: {quantile_estimates[i]}")
-    25th percentile: 3.1505820231763066
+    25th percentile: 3.1505820231763066 # may vary
```
Comment (ev-br) [inline scipy/stats/_mstats_extras.py:75]: Great idea to highlight the problematic line!

File: scipy/linalg/_decomp_schur.py (hunk 1 above)
Comment (tylerjereddy) [reply scipy/linalg/_decomp_schur.py:105]: I didn't know that, but I did see some fairly striking differences:

```
File "build-install/lib/python3.11/site-packages/scipy/linalg/_decomp_schur.py", line 102, in schur
Failed example:
    T2
Expected:
    array([[ 2.65896708, -1.22839825+1.32378589j,  0.42590089+1.51937378j],
           [ 0.        , -0.32948354+0.80225456j, -0.59877807+0.56192146j],
           [ 0.        ,  0.                    , -0.32948354-0.80225456j]])
Got:
    array([[ 2.65896708-3.33066907e-16j, -1.58397876+8.67399929e-01j,
             0.42590083+1.51937380e+00j],
           [ 0.        +0.00000000e+00j, -0.32948354+8.02254558e-01j,
            -0.39035561+7.22435738e-01j],
           [ 0.        +0.00000000e+00j,  0.        +0.00000000e+00j,
            -0.32948354-8.02254558e-01j]])
```

**Comment by lucascolley** [comment]
on this branch I get the following refguide errors:

<details>

```
=======================
scipy.cluster.hierarchy
=======================

scipy.cluster.hierarchy.centroid
--------------------------------

File "build-install/lib/python3.10/site-packages/scipy/cluster/hierarchy.py", line 562, in centroid
Failed example:
    Z
Expected:
    array([[ 0.        ,  1.        ,  1.        ,  2.        ],
           [ 3.        ,  4.        ,  1.        ,  2.        ],
           [ 9.        , 10.        ,  1.        ,  2.        ],
           [ 6.        ,  7.        ,  1.        ,  2.        ],
           [ 2.        , 12.        ,  1.11803399,  3.        ],
           [ 5.        , 13.        ,  1.11803399,  3.        ],
           [ 8.        , 15.        ,  1.11803399,  3.        ],
           [11.        , 14.        ,  1.11803399,  3.        ],
           [18.        , 19.        ,  3.33333333,  6.        ],
           [16.        , 17.        ,  3.33333333,  6.        ],
           [20.        , 21.        ,  3.33333333, 12.        ]])
Got:
    array([[ 0.        ,  1.        ,  1.        ,  2.        ],
           [ 3.        ,  4.        ,  1.        ,  2.        ],
           [ 9.        , 10.        ,  1.        ,  2.        ],
           [ 6.        ,  7.        ,  1.        ,  2.        ],
           [ 2.        , 12.        ,  1.11803399,  3.        ],
           [ 5.        , 13.        ,  1.11803399,  3.        ],
           [ 8.        , 15.        ,  1.11803399,  3.        ],
           [11.        , 14.        ,  1.11803399,  3.        ],
           [16.        , 18.        ,  3.33333333,  6.        ],
           [17.        , 19.        ,  3.33333333,  6.        ],
           [20.        , 21.        ,  3.33333333, 12.        ]])


File "build-install/lib/python3.10/site-packages/scipy/cluster/hierarchy.py", line 582, in centroid
Failed example:
    fcluster(Z, 0.9, criterion='distance')
Expected:
    array([ 7,  8,  9, 10, 11, 12,  1,  2,  3,  4,  5,  6], dtype=int32)
Got:
    array([ 1,  2,  3,  7,  8,  9,  4,  5,  6, 10, 11, 12], dtype=int32)


File "build-install/lib/python3.10/site-packages/scipy/cluster/hierarchy.py", line 584, in centroid
Failed example:
    fcluster(Z, 1.1, criterion='distance')
Expected:
    array([5, 5, 6, 7, 7, 8, 1, 1, 2, 3, 3, 4], dtype=int32)
Got:
    array([1, 1, 2, 5, 5, 6, 3, 3, 4, 7, 7, 8], dtype=int32)


File "build-install/lib/python3.10/site-packages/scipy/cluster/hierarchy.py", line 586, in centroid
Failed example:
    fcluster(Z, 2, criterion='distance')
Expected:
    array([3, 3, 3, 4, 4, 4, 1, 1, 1, 2, 2, 2], dtype=int32)
Got:
    array([1, 1, 1, 3, 3, 3, 2, 2, 2, 4, 4, 4], dtype=int32)


============
scipy.linalg
============

scipy.linalg.schur
------------------

File "build-install/lib/python3.10/site-packages/scipy/linalg/_decomp_schur.py", line 102, in schur
Failed example:
    T2
Expected:
    array([[ 2.65896708, -1.22839825+1.32378589j,  0.42590089+1.51937378j],
           [ 0.        , -0.32948354+0.80225456j, -0.59877807+0.56192146j],
           [ 0.        ,  0.                    , -0.32948354-0.80225456j]])
Got:
    array([[ 2.65896708-3.33066907e-16j, -1.58397876+8.67399929e-01j,
             0.42590098+1.51937375e+00j],
           [ 0.        +0.00000000e+00j, -0.32948354+8.02254558e-01j,
            -0.39035554+7.22435776e-01j],
           [ 0.        +0.00000000e+00j,  0.        +0.00000000e+00j,
            -0.32948354-8.02254558e-01j]])


==============
scipy.optimize
==============

scipy.optimize.shgo
-------------------

File "build-install/lib/python3.10/site-packages/scipy/optimize/_shgo.py", line 424, in shgo
Failed example:
    res
Expected:
     message: Optimization terminated successfully.
     success: True
         fun: 29.894378159142136
        funl: [ 2.989e+01]
           x: [ 6.355e-01  1.137e-13  3.127e-01  5.178e-02]
          xl: [[ 6.355e-01  1.137e-13  3.127e-01  5.178e-02]]
         nit: 1
        nfev: 142
       nlfev: 35
       nljev: 5
       nlhev: 0
Got:
     message: Optimization terminated successfully.
     success: True
         fun: 29.894378159168856
        funl: [ 2.989e+01]
           x: [ 6.355e-01  0.000e+00  3.127e-01  5.178e-02]
          xl: [[ 6.355e-01  0.000e+00  3.127e-01  5.178e-02]]
         nit: 1
        nfev: 133
       nlfev: 26
       nljev: 5
       nlhev: 0


==================
scipy.stats.mstats
==================

scipy.stats.mstats.hdquantiles
------------------------------

File "build-install/lib/python3.10/site-packages/scipy/stats/_mstats_extras.py", line 73, in hdquantiles
Failed example:
    for i, quantile in enumerate(probabilities):
        print(f"{int(quantile * 100)}th percentile: {quantile_estimates[i]}")
Expected:
    25th percentile: 3.1505820231763066
    50th percentile: 5.194344084883956
    75th percentile: 7.430626414674935
Got:
    25th percentile: 3.150582023176307
    50th percentile: 5.194344084883956
    75th percentile: 7.430626414674935


ERROR: refguide or doctests have errors
```

</details>

File: scipy/linalg/_decomp_schur.py (hunk 1 above)
Comment (ev-br) [reply scipy/linalg/_decomp_schur.py:105]: Could you double-check that the result on your system is a correct Schur decomp? Basically, that indeed `A = Z T Z^H` modulo floating-point noise? 

My suspicion is that the root of the problem is that the Schur decomp is not unique, and that the example and your system just return two different, perfectly legal, factorizations. If that's indeed the case, a single `# may vary` incantation anywhere in the `want` text would be enough to paper over the hard failure.  (or `# random` -- this is functionally equivalent but maybe not very appropriate for readability here? --- both are essentially a `# doctest: +SKIP`, only apply to `want` not `source` in the doctest parlance) . 

A long-term fix would be to adjust the example to either do not list individual matrices or add some text (maybe also a reference?) that indeed there is variability and that's OK. But that's probably a separate doc enhancement anyway.

**Comment by ev-br** [comment]
> on this branch I get the following refguide errors

All of these look to me as too-strict doctests, which rely on implementation details of whatever setup the original author had:  is the exact ordering of centroids guaranteed? (I doubt it). Do we guarantee the exact number of evaluations in shgo? Again, I doubt it. 

The `hdquantiles` failure is a bit different though: if we do print things, why do we print all digits? Either print two max three sig figs or output them as they are without any formatting. The technical issue here is that refguide-check fails to fish numbers out the sea of text and falls back to the vanilla doctest --- which compares strings symbol by symbol, with a not-entirely-unexpected outcome.

File: scipy/linalg/_decomp_schur.py (hunk 1 above)
Comment (tylerjereddy) [reply scipy/linalg/_decomp_schur.py]: I pushed in the simplification to use a single `# may vary`. If you want me to check the linalg thing I may need to be spoon fed how to do that, unfortunately. I suppose I might hope that the full test suite wouldn't pass if I had a fundamental local issue.

I'm happy for Lucas to push in the refguide shims he needs on his local machine as well, if you want.

**Review by ev-br — APPROVED** [review]
LGTM. 
Further improvements can go either here or to a follow-up, whichever works.

**Comment by lucascolley** [comment]
> Further improvements can go either here or to a follow-up, whichever works.

Get this in 👍  I can open an issue with my local errors later.

**Comment by ev-br** [comment]
Thank you Tyler, Lucas.

**Pull request merged**